operators/pkg/instctrl/cloudinit.go

```diff
@@ -45,47 +46,82 @@ const (
 func (r *InstanceReconciler) EnforceCloudInitSecret(ctx context.Context) error {
 	log := ctrl.LoggerFrom(ctx)
 
-	// Retrieve the WebDav credentials.
-	user, password, err := r.GetWebDavCredentials(ctx)
-	if err != nil {
-		log.Error(err, "unable to get webdav credentials")
-		return err
-	}
-	log.V(utils.LogDebugLevel).Info("webdav credentials correctly retrieved")
+	var retErr error
+	var server string
+	var path string
 
 	// Retrieve the public keys
-	publicKeys, err := r.GetPublicKeys(ctx)
-	if err != nil {
-		log.Error(err, "unable to get public keys")
-		return err
+	publicKeys, retErr := r.GetPublicKeys(ctx)
+	if retErr != nil {
+		log.Error(retErr, "unable to get public keys")
+		return retErr
 	}
 	log.V(utils.LogDebugLevel).Info("public keys correctly retrieved")
 
-	userdata, err := forge.CloudInitUserData(r.ServiceUrls.NextcloudBaseURL, user, password, publicKeys)
-	if err != nil {
-		log.Error(err, "unable to marshal secret content")
-		return err
+	if _, server, path, retErr = r.GetNfsSpecs(ctx); retErr != nil {
+		log.Error(retErr, "unable to retrieve nfs volume dns name and path")
+		return retErr
+	}
+
+	userdata, retErr := forge.CloudInitUserData(server, path, publicKeys)
+	if retErr != nil {
+		log.Error(retErr, "unable to marshal secret content")
+		return retErr
 	}
 
 	// Enforce the cloud-init secret presence
 	instance := clctx.InstanceFrom(ctx)
 	secret := corev1.Secret{ObjectMeta: forge.ObjectMeta(instance)}
-	res, err := ctrl.CreateOrUpdate(ctx, r.Client, &secret, func() error {
-		secret.SetLabels(forge.InstanceObjectLabels(secret.GetLabels(), instance))
+	res, retErr := ctrl.CreateOrUpdate(ctx, r.Client, &secret, func() error {
 		secret.Data = map[string][]byte{UserDataKey: userdata}
 		secret.Type = corev1.SecretTypeOpaque
 		return ctrl.SetControllerReference(instance, &secret, r.Scheme)
 	})
 
-	if err != nil {
-		log.Error(err, "failed to enforce cloud-init secret", "secret", klog.KObj(&secret))
-		return err
+	if retErr != nil {
+		log.Error(retErr, "failed to enforce cloud-init secret", "secret", klog.KObj(&secret))
+		return retErr
```

Comment:
Would revert

Reply:
Yes, in general we're talking about diffs of code, relatively to how it was before your commits. You can see diffs in the "files changed" tab of the PR. You can resolve conversations directly when you handled them, in case no further information/discussion is needed.